thrust/thrust/detail/complex.inl

```diff
@@ -0,0 +1,232 @@
+#pragma once
+
+#include <thrust/complex.h>
+#include <thrust/detail/config.h>
+#include <thrust/type_traits/is_trivially_relocatable.h>
+
+THRUST_NAMESPACE_BEGIN
+
+/* --- Copy Constructor --- */
+
+template <typename T>
+template <typename U>
+__host__ __device__ complex<T>::complex(const ::cuda::std::complex<U> &rhs)
+    : ::cuda::std::complex<T>(rhs)
+{}
+
+/* --- Copy Asignment --- */
+
+template <typename T>
+template <typename U>
+__host__ __device__
+  typename detail::disable_if<detail::is_same<T, U>::value,
+                              complex<typename detail::promoted_numerical_type<T, U>::type>>::type &
+  complex<T>::operator=(const ::cuda::std::complex<U> &z)
+{
+  this->real(T(z.real()));
+  this->imag(T(z.imag()));
+  return *this;
+}
+
+/* --- Equality Operators --- */
+
+template <typename T0, typename T1>
+__host__ __device__ typename detail::disable_if<detail::is_same<T0, T1>::value, bool>::type
+operator==(const complex<T0> &x, const complex<T1> &y)
```

Comment:
I now I initially removed the files with implementations of the different operators / functions.

That said, I would much appreciate, if we could rather put the additional overloads that thrust provides into the old files and remove anything we do not need anymore

Reply:
Will be done in a later commit.